### PR TITLE
Allow pixel snap for rotated objects (issue 14496)

### LINKED ISSUE #14496: Snapping doesnt work after rotate node
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Arch Linux, Godot 3 beta

**Issue description:**

Snapping doesnt work after rotate node.
It works in Godot 2.1

**Steps to reproduce:**
Just create Node2D and rotate.

**Link to minimal example project:**
Just create new project.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #14496

@groud The change here is to check for `rotation == 0` only during forced snapping and allow it always if it's explicitly set

* old condition: ("snap pixel is set" OR "force snap pixel") AND rotation == 0
* new condition: "snap pixel is set" OR (force snap pixel AND rotation == 0)

For the record, in Godot 2.1 this code is:

```
Vector2 CanvasItemEditor::snap_point(Vector2 p_target, Vector2 p_start) const {
	if (snap_grid) {
		p_target.x = _snap_scalar(snap_offset.x, snap_step.x, snap_relative, p_target.x, p_start.x);
		p_target.y = _snap_scalar(snap_offset.y, snap_step.y, snap_relative, p_target.y, p_start.y);
	}
	if (snap_pixel)
		p_target = p_target.snapped(Size2(1, 1));

	return p_target;
}
```

### CODE REVIEW

**Comment by poke1024** [comment]
Does this really need a new parameter? I suspect that `rotation` should never get considered for a mode if the mode comes from an explicitly set `p_modes`.

Correct me if I'm wrong, but the whole `rotation` check seems to be something very special for the one border case when dragging anchors, the one single call to `snap_point` that sets `p_forced_mode` to some non-0 value; all other calls to `snap_point` don't use `p_forced_mode`.

**Comment by groud** [comment]
The rotation == 0 here is to avoid pixel (or grid) snaping when resizing a node that is rotated: we don't want it to snap to the pixel/grid when resizing a rotated node, it would be annoying. 
The p_forced_mode, unlike p_modes, are just used to bypass the editor snapping parameters.

This function is meant to be a all-in-one tool for snapping, it's supposed to be a public function at one point. Thus it should be generic enough, not only fit our specific use case.

So basically, we want to be able to know if we want to take into account the rotation (for nodes resizing) or not (for moving Node2D). This requires either a new parameter.

**Comment by reduz** [comment]
What's the status of this?

**Comment by groud** [comment]
I think the issue this PR was trying to solve is now solvd thanks to the editor rework. Simple Node2D (without a rect) can ben snapped even if rotated.
So I guess the PR can be closed.
